src/System.Private.CoreLib/src/System/String.cs

```diff
@@ -2717,24 +2717,30 @@ public String Replace(char oldChar, char newChar)
         {
             if (oldChar == newChar)
                 return this;
-            
+
             unsafe
             {
-                int firstFoundIndex = -1;
+                int remainingLength = Length;
+                char* pSrc;
```

Comment:
It would be better if there are inside the fixed block. It is fit with the structure of the code - `pChars` is there twice too, and it is a good defensive programming practice to not let the pointers be visible outside the fixed block that is pinning them.